charms/worker/k8s/src/token_distributor.py

```diff
@@ -332,30 +400,38 @@ def allocate_tokens(
                 # if the unit leaves. Let's create a cache in
                 # our app's session of this data.
                 log.info(
-                    "Completed token allocation of %s with unit=%s:%s (%s)",
-                    token_type.value,
+                    "Completed %s token allocation of %s with unit=%s:%s (%s)",
+                    token_strategy.name.title(),
+                    token_type.name.title(),
                     relation.name,
                     unit.name,
                     node,
                 )
                 self.update_node(relation, unit, f"joined-{node}")
-                if revoke_on_join:
+                if tokenizer.revoke_on_join:
```

Comment:
some tokens may need to be revoked once used for joining.  Let's push that into the `TokenManager` logic